packages/react-client/src/ReactFlightClient.js

```diff
@@ -379,9 +402,13 @@ function initializeModelChunk<T>(chunk: ResolvedModelChunk<T>): void {
       blockedChunk.value = null;
       blockedChunk.reason = null;
     } else {
+      const resolveListeners = cyclicChunk.value;
```

Comment:
Discovering a cycle doesn't add to the `.deps` so we might end up at zero. The inner property listens for this model to complete so that it can update its property to set the cycle.